server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1420,19 +1426,22 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
                         ShardRouting primary = indexRoutingTable.shard(i).primaryShard();
                         if (primary == null || !primary.assignedToNode()) {
                             builder.put(shardId,
-                                new SnapshotsInProgress.ShardSnapshotStatus(null, ShardState.MISSING, "primary shard is not allocated"));
+                                new SnapshotsInProgress.ShardSnapshotStatus(null, ShardState.MISSING, "primary shard is not allocated",
```

Comment:
Passing `null` for the generations for now since we don't know them when starting a shard snapshot yet. #46250 will make use of the ability to pass the shard-generation via the CS by determining the initial shard gen values from the repositories' `RepositoryData` here.